compiler/rustc_sanitizers/src/cfi/typeid/itanium_cxx_abi/encode.rs

```diff
@@ -85,7 +88,10 @@ fn encode_args<'tcx>(
                     s.push_str(&encode_ty(tcx, ty, dict, options));
                 }
                 GenericArgKind::Const(c) => {
-                    s.push_str(&encode_const(tcx, c, dict, options));
+                    let n = n + (has_erased_self as usize);
+                    let ct_ty =
+                        tcx.type_of(def_generics.param_at(n, tcx).def_id).instantiate_identity();
```

Comment:
Could this ty resolution be done in encode_const?

Reply:
This instantiation is different.